File: rei-do-chopp-site/src/app/users/models/requests/user-list.request.ts
import { PaginationRequest } from "../../../shared/models/requests/pagination.request";

export class UserListRequest extends PaginationRequest
{
  Id: number;
  Name: string;
  Active?: boolean;
  RoleIds: number[];

  constructor(params: Partial<UserListRequest>)
  {
    super(params);
    this.Id = params.Id || null;
    this.Name = params.Name || ""
    this.Active = params.Active ?? null;
    this.RoleIds = params.RoleIds || [];
  }
}

File: rei-do-chopp-api/ReiDoChopp.DataTransfer/Users/Requests/UserListRequest.cs
using System;
using ReiDoChopp.Domain.Utils.Enums;
using ReiDoChopp.DataTransfer.Pagination.Responses;
using ReiDoChopp.DataTransfer.Pagination.Requests;

namespace ReiDoChopp.DataTransfer.Users.Requests
{
    public class UserListRequest : PaginationRequest
    {
        public int? Id { get; set; }
        public string Name { get; set; }
        public bool? Active { get; set; }
        public int[] RoleIds { get; set; }

        public UserListRequest() : base(pageSize: 10, ordenationField: "Id", ordenationType: OrdenationTypeEnum.ASC) {}
    }
}